### DISCUSSION TITLE
Get the statistics of radio sleep time

### DISCUSSION THREAD

**Discussion by Irving-cl** [openthread/openthread#8820] [Ideas]
Recently I'm trying to investigate to what extent can Thread Low Power feature benefit real use cases. So I'm trying to get some data to reflect how much power can be saved. While we can use an ammeter to doing so, I think getting a metrics from SW should make things easier. (Especially when we test with multiple end devices) And radio sleep time itself should be a good metrics to reflect if a low power solution is successful.

I want to have some APIs like this:
```
// This API starts counting the radio sleep time of the device.
otError otRadioStartCountSleepTime();

// This API gets the radio sleep since last `Start` was called
uint64_t otRadioGetSleepTime();

// This API stops counting the radio sleep time of the device and clear the number
otError otRadioStopCountSleepTime();
```

I'm thinking if we can add this to OpenThread core and let `SubMac` or `Radio` layer to count this or we can add a new platform API. I think adding the API in platform will provide more precise number.

Any thoughts? @abtink @edmont 

Is it possible that we implement this in nrf528xx platform? @edmont

**Comment by edmont** [comment]
Hi @Irving-cl, this seems like a good idea to me and shouldn't be too difficult to implement.

However, not sure if radio activity would be the only relevant thing impacting power consumption. For instance, based on real ammeter data, I've observed that the CPU time used for scheduling delayed reception in CSL implementation has some relevant impact on the average energy usage.

So maybe it would be better to consider some struct with some more fields apart from radio sleep time?
